src/DDM.jl

```diff
@@ -125,230 +226,493 @@ function _large_time_pdf(u::T, z::T, K::Int) where {T<:Real}
 
-    return _P_upper(ν, α, z) + sign(ν) * ((cdf(Normal(), -sign(ν) * (α*z+ν*(t-τ))/sqt) -
-             _exp_pnorm(-2*ν*α*z, sign(ν) * (α*z-ν*(t-τ)) / sqt)) + S1 + S2)
-end
+"""
+# Simpson's Method two dimentional case
+function _simpson_2D(x::Real, ν::Real, η::Real, α::Real, z::Real, τ::Real, ϵ::Real, lb_z::Real, ub_z::Real, n_sz::Int, lb_t::Real, ub_t::Real, n_st::Int)
 
-# Zero drift version
-function _Fs0_lower(d::DDM{T}, K::Int, t::Real) where {T<:Real}
-    (_, α, τ, z) = params(d)
-    F = zero(T)
-    K_series = K:-1:0
-    for k in K_series
-        F -= (cdf(Distributions.Normal(), (-2*k - 2+z) * α / sqrt(t-τ)) + cdf(Distributions.Normal(), (-2*k -z) * α / sqrt(t-τ)))
-    end
-    return 2*F
-end
-# Number of terms required for large time representation
-function _K_large(d::DDM{T}, t::Real; ϵ::Real = 1.0e-12) where {T<:Real}
-    (ν, α, τ, z) = params(d)
-    x = t-τ
-    sqrtL1 = sqrt(1/x) * α/π
-    sqrtL2 = sqrt(max(1, -2/x*α*α/π/π * (log(ϵ*π*x/2 * (ν*ν + π*π/α/α)) + ν*α*z + ν*ν*x/2)))
-    return ceil(Int, max(sqrtL1, sqrtL2))
-end
+    ht = (ub_t-lb_t)/n_st
+    S = _simpson_1D(x, ν, η, α, z, τ, ϵ, lb_z, ub_z, n_sz, 0, 0, 0)
 
-# Number of terms required for small time representation
-function _K_small(d::DDM{T}, t::Real; ϵ::Real = 1.0e-12) where {T<:Real}
-    (ν, α, τ, z) = params(d)
-    if abs(ν) < sqrt(eps(T))
-        return ceil(Int, max(0, z/2 - sqrt(t-τ)/(2*α) * quantile(Normal(), max(0, min(1, ϵ/(2-2*z))))))
-    end
-    if ν > 0
-        return _K_small(DDM(-ν, α, τ, z), t; ϵ = exp(-2*α*z*ν)*ϵ)
-    end
-    S2 = z - 1 + 1/(2*ν*α) * log(ϵ/2 * (1-exp(2*ν*α)))
-    S3 = (0.535 * sqrt(2*(t-τ)) + ν*(t-τ) + α*z)/(2*α)
-    S4 = z/2 - sqrt(t-τ)/(2*α) * quantile(Normal(), max(0, min(1, ϵ*α/(0.3 * sqrt(2*π*(t-τ))) * exp(ν^2*(t-τ)/2 + ν*α*z) ))) 
-    return ceil(Int, max(0, S2, S3, S4))
-end
-# Probability for absorption at upper barrier
-function _P_upper(ν::T, α::T, z::T) where {T<:Real}
-    e = exp(-2 * ν * α * (1-z))
-    if isinf(e)
-        return 1.0
-    end
-    if abs(e-1) < sqrt(eps(T))
-        return 1-z
-    end
-    return (1-e)/(exp(2*ν*α*z) - e)
-end
+    t_tag = 0
+    y = 0 
+    for i_t in 1:n_st
+        t_tag = lb_t + ht * i_t
+        y = _simpson_1D(x, ν, η, α, z, t_tag, ϵ, lb_z, ub_z, n_sz, 0, 0, 0)
 
-# Calculates exp(a) * pnorm(b) using an approximation by Kiani et al. (2008)
-function _exp_pnorm(a::T, b::T) where {T<:Real}
-    r = exp(a) * cdf(Distributions.Normal(), b)
-    if isnan(r) && b < -5.5
-        r = (1/sqrt(2)) * exp(a - b^2/2) * (0.5641882/(b^3) - 1/(b * sqrt(π))) 
+        if isodd(i_t)
+            S += 4 * y
+        else
+            S += 2 * y
+        end
     end
-    return r
+
+    S = S - y  # the last term should be f(b) and not 2*f(b) so we subtract y
+    S = S / (ub_t - lb_t)
+
+    return ht * S / 3
+
 end
 
+logpdf(d::DDM, choice, rt; ϵ::Real = 1.0e-12) = log(pdf(d, choice, rt; ϵ))
+
+
+"""
```

Comment:
Can you please comment out the doc strings for cdf? I think it will show up in the documentation even though there is not a method for DDM.